lib/data/wrapper.js

```diff
@@ -139,18 +142,21 @@ function _put(cipherBundle, value, valueSize,
     }
 
     return client.put(writeStream, valueSize, keyContext,
-                      log.getSerializedUids(), (err, key) => {
-                          if (err) {
-                              log.error('put error from datastore',
-                                        { error: err, implName });
-                              return cb(errors.InternalError);
-                          }
-                          const dataRetrievalInfo = {
-                              key,
-                              dataStoreName: implName,
-                          };
-                          return cb(null, dataRetrievalInfo, hashedStream);
-                      });
+        log.getSerializedUids(), (err, key) => {
+            if (err) {
+                log.error('put error from datastore',
+                        { error: err, implName });
+                if (err.httpCode === 408) {
+                    return cb(errors.RequestTimeout);
```

Comment:
IncompleteBody is a more accurate description of the error. We timeout the connection because the client has not sent what they specified in the Content-Length HTTP header